### DISCUSSION TITLE
Reading room names from map - custom Roborock integration

### DISCUSSION THREAD

**Discussion by znanywawa** [PiotrMachowski/lovelace-xiaomi-vacuum-map-card#586] [Ideas]
Hi,

You made a great job with this card :)
With new update of Roborock custom integration (https://github.com/humbertogontijo/homeassistant-roborock) there is ability to read room names from Roborock app.

Is it possible to add on map room names (now we have to define possitions X and Y possitions and provide room number to send Roborock vacuum to correct room, but maybe it is possible to use data from Roborock integration?

**Comment by PiotrMachowski** [comment]
Could you provide some more data about this functionality? How does it specify room names?

**Reply by znanywawa** [reply]
Current version of this custom integration adds couple new entities.
One of them is related to room where vacuum is for now, in my case entity name is: sensor.roborock_s7_current_room
So I assume, it may be possible to somehow combine this entity with map entity for Roborock complete native map instead of defining rooms in map card configuration.

As well, new version does provides name of map that is currently used, entity is: sensor.roborock_s7_current_map_selected
So maybe... it is possible to support multiple maps by switching them via map card (just thinking...)

**Reply by PiotrMachowski** [reply]
Rooms in maps configuration are already created automatically using attributes of the camera entity. Room names are already supported in the generation process. I have decided to make it this way to allow users to customize maps details (rooms' icons, names and outlines). Do you see a way to improve this process?

Switching maps is already supported, check out a discussion in [this issue](https://github.com/PiotrMachowski/lovelace-xiaomi-vacuum-map-card/issues/248)

**Reply by znanywawa** [reply]
Yes, but rooms generated by camera still requires some adjustments (at least I had to make them) as some rooms were generated overcoming to others. Also, splitting rooms in graphical way (by Roborock app) are easier and more user friendly. So... (just thinking), maybe it is possible to add switch for card configuration, where user can define does he wants to use Roborock rooms or he wants to more customizable ones generated by camera (of course, then there is question does icons assignment will be possible for rooms from Roborock app if not, it will require to pring room name on map card).

**Reply by PiotrMachowski** [reply]
Buuuuut ones generated by the camera _are_ the same rooms as in the Roborock app. The problem is in the detection of room boundaries. The integration currently calculates a bounding rectangle for every room instead, which is a much easier task.

By the way, icons can also be used in the generated config if they are provided by the camera.

**Reply by znanywawa** [reply]
Yes, and those boundaries are something that may drive crazy for beginner.
I do understand that as You and many users are able to manipulate in configuration on their own, I was just thinking about new users and more friendly for them (as in my vision: they could choose do they want to use rooms camera generated or imported from Roborock while using Roborock integration).

It was only an idea, and I thought that maybeee it may be easy to implement, but if not, just drop it :)

**Reply by PiotrMachowski** [reply]
But what can be imported from the Roborock? As far as I know everything that is imported is already used by the card. Do you have some more details?

**Reply by znanywawa** [reply]
No I dont, I think that we can drop it :)

**Reply by PiotrMachowski** [reply]
The card already supports autogenerating rooms with a proper outline, so the rest has to be provided by the integration :)
